gitmanager.py

```diff
@@ -306,20 +306,56 @@ def remove_from_blacklist(cls, item, username, blacklist_type="", code_privilege
     @classmethod
     def merge_pull_request(cls, pr_id, comment=""):
         response = requests.get("https://api.github.com/repos/{}/pulls/{}".format(GlobalVars.bot_repo_slug, pr_id))
-        if not response:
+        files_list = requests.get("https://api.github.com/repos/{}/pulls/{}/files"
+                                  .format(GlobalVars.bot_repo_slug, pr_id))
+        if not response or not files_list:
             raise ConnectionError("Cannot connect to GitHub API")
         pr_info = response.json()
+        files_info = files_list.json()
         if pr_info["user"]["login"] != "SmokeDetector":
             raise ValueError("PR #{} is not created by me, so I can't approve it.".format(pr_id))
         if "<!-- METASMOKE-BLACKLIST" not in pr_info["body"]:
             raise ValueError("PR description is malformed. Blame a developer.")
         if pr_info["state"] != "open":
             raise ValueError("PR #{} is not currently open, so I won't merge it.".format(pr_id))
+        pr_string = pr_info["title"]
+        file_modified = files_info[0]["filename"]
         ref = pr_info['head']['ref']
+        requested_pr_string = regex.match(r".*?: \S+ (.*?)(?:\Z|\s*\(\?#)", pr_string).group(1)
+        username_of_pr_requester = regex.match(r".*(?=:)", pr_string).group(0)
+        if file_modified == "blacklisted_websites.txt":
+            blacklist_type = Blacklist.WEBSITES
+        elif file_modified == "bad_keywords.txt":
+            blacklist_type = Blacklist.WEBSITES
+        elif file_modified == "blacklisted_usernames.txt":
+            blacklist_type = Blacklist.USERNAMES
+        elif file_modified == "blacklisted_numbers.txt":
+            blacklist_type = Blacklist.NUMBERS
+        elif file_modified == "watched_keywords.txt":
+            blacklist_type = Blacklist.WATCHED_KEYWORDS
+        elif file_modified == "watched_numbers.txt":
+            blacklist_type = Blacklist.WATCHED_NUMBERS
+        else:
+            raise CmdException('GitManager: blacklist is not recognized. Blame a developer.')
+        try:
+            blacklister = Blacklist(blacklist_type)
+            blacklist_file_name = blacklist_type[0]
+            item_to_blacklist = requested_pr_string
+            if blacklist_type in {Blacklist.WATCHED_KEYWORDS, Blacklist.WATCHED_NUMBERS}:
+                op = 'watch'
+                item = item_to_blacklist
+                item_to_blacklist = "\t".join([now, username_of_pr_requester, item])
+            else:
+                op = 'blacklist'
+                item = item_to_blacklist
 
+            exists, line = blacklister.exists(item_to_blacklist)
+            if exists:
+                raise RuntimeError('Already {}ed on line {} of {}'.format(op, line, blacklist_file_name))
+        except Exception:
+            raise CmdException("An error occured")
```

Comment:
Sorry I missed this last time around — but this is almost definitely not what you want to do. If *any* error occurs within the function (including your new “already actioned” error), this will discard the error and only produce the response “An error occurred”.

Instead, you probably want to:

1. Delete this except block. That way, the error will be handled by chatcommunicate which will log the error and post the “I hit an error [...] run `!!/errorlogs` for details.”
2. Change the `RuntimeError` on line 354 to a `CmdException`. That way the “already actioned” message will be posted to chat directly (instead of going to errorlogs)